### PR TITLE
allow renditions based on ratio

### PR DESCRIPTION
**Pull request by vied12**

### CODE REVIEW

File: superdesk/media/crop.py [hunk 1]
```diff
@@ -109,17 +110,23 @@ def _validate_aspect_ratio(self, crop, doc):
...
-
-        if width < crop_width or height < crop_height:
-            raise SuperdeskApiError.badRequestError(
-                message='Wrong crop size. Minimum crop size is {}x{}.'.format(crop['width'], crop['height']))
-
-        doc_ratio = round(width / height, 1)
-        spec_ratio = round(crop_width / crop_height, 1)
-        if doc_ratio != spec_ratio:
-            raise SuperdeskApiError.badRequestError(message='Wrong aspect ratio!')
+        assert (crop.get('width') or crop.get('height')) or crop.get('ratio')
```
Comment (mugurrus) [inline superdesk/media/crop.py]: Better raise an exception (e.g. SuperdeskApiError) that transmits more info

File: superdesk/media/crop.py [hunk 2]
```diff
@@ -109,17 +110,23 @@ def _validate_aspect_ratio(self, crop, doc):
...
+                doc_ratio = round(width / height, 1)
+                spec_ratio = round(expected_crop_width / expected_crop_height, 1)
+                if doc_ratio != spec_ratio:
+                    raise SuperdeskApiError.badRequestError(message='Wrong aspect ratio!')
+        elif crop.get('ratio'):
+            ratio = crop.get('ratio')
+            if type(crop.get('ratio')) not in [int, float]:
+                ratio = ratio.split(':')
+                ratio = int(ratio[0]) / int(ratio[1])
+            assert(abs((width / height) - ratio) <= 0.01)
```
Comment (mugurrus) [inline superdesk/media/crop.py]: Same here

**Comment by mugurrus** [comment]
:+1:

File: superdesk/media/renditions.py [hunk 3]
```diff
@@ -92,15 +141,23 @@ def resize_image(content, format, size, keepProportions=True):
     if keepProportions:
         width, height = img.size
         new_width, new_height = size
-        x_ratio = width / new_width
-        y_ratio = height / new_height
-        if x_ratio > y_ratio:
-            new_height = int(height / x_ratio)
+        assert new_width is not None or new_height is not None
```
Comment (actionless) [inline superdesk/media/renditions.py]: and here one more

**Comment by petrjasek** [comment]
:+1:

**Pull request merged**